### PR TITLE
Developer work-experience block

### PR DESCRIPTION
**Pull request by timazubar**
- add work-experience as projects in developer model
- add work-experience page, where projects are displayed
- add work-experience section with project-cards in developer-resume-full template

### CODE REVIEW

File: src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-work-experience.component.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<app-profile-section-header [title]="'Work experience'" (edit)="onEditClick()"></app-profile-section-header>
+
+<div class="card d__flex d__flex--direction-column p__20" *ngFor="let project of (developer$ | async).projects">
+  <span class="text text__title" *ngIf="!isEdit">{{ project.title }}</span>
```
Comment (tergog) [inline src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-work-experience.component.html:4]:
```suggestion
<ng-container *ngIf="!isEdit">
  <span class="text text__title">{{ project.title }}</span>
  <p class="text text__md">{{ project.description }}</p>
</ng-container>
```

**Review by tergog — APPROVED** [review]

File: src/app/inner-pages/dev-pages/dev-profile/dev-categories-and-skills/dev-categories-and-skills.component.html [hunk 2]
```diff
@@ -1,64 +1,3 @@
...
-      </mat-option>
-    </mat-autocomplete>
-  </mat-form-field>
-
-  <div class="text__right" *ngIf="isEdit">
-    <app-button [type]="'transparent'" [text]="'Cancel'" (buttonClick)="onEditClick()"></app-button>
-    <app-button [text]="'Save'" (buttonClick)="onSaveClick()"></app-button>
-  </div>
-</div>
+<app-edit-form [sectionName]="'categoriesAndSkills'" [isEdit]="isEdit" (editToggle)="onEditClick()"></app-edit-form>
```
Comment (timazubar) [inline src/app/inner-pages/dev-pages/dev-profile/dev-categories-and-skills/dev-categories-and-skills.component.html:3]: sectionNames to enum.
edit form naming

File: src/app/inner-pages/dev-pages/dev-profile/dev-profile-sections/dev-profile-sections.component.ts [hunk 3]
```diff
@@ -32,10 +31,9 @@ export class DevProfileSectionsComponent implements OnInit {
     private notificationsService: NotificationsService
   ) { }
 
-  ngOnInit(): void {
-  }
+  ngOnInit(): void {}
 
-  onSaveClick(userInfo) {
+  public onSaveClick(userInfo) {
```
Comment (timazubar) [inline src/app/inner-pages/dev-pages/dev-profile/dev-profile-sections/dev-profile-sections.component.ts:36]: add return type

File: src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-project-card/dev-project-card.component.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+                  [selectable]="false"
+                  [removable]="isEdit"
+                  (removed)="onTechnologyRemove(technology)">
+          {{ technology.name }}
+        </mat-chip>
+      </mat-chip-list>
+  </ng-container>
+
+  <ng-container *ngIf="isEdit">
+
```
Comment (timazubar) [inline src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-project-card/dev-project-card.component.html:18]: ngIf on form

File: src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-project-card/dev-project-card.component.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    { name: 'HTML5', value: 5 },
+    { name: 'AngularJS', value: 6 },
+    { name: 'Angular 9', value: 7 },
+    { name: 'Angular 10', value: 8 },
+    { name: 'Angular 7', value: 9 },
+    { name: 'Angular 8', value: 10 },
+    { name: 'Angular 2+', value: 11 },
+  ];
+
+  @Input() project: DevProject;
```
Comment (timazubar) [inline src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-project-card/dev-project-card.component.ts:40]: move inputs/outputs on top

File: src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-project-card/dev-project-card.component.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+  private initForm(): void {
+    this.form = new FormGroup({
+      title: new FormControl('', []),
+      description: new FormControl('', []),
+      technologies: new FormControl([], []),
+      link: new FormControl('', [])
+    });
+  }
+
```
Comment (timazubar) [inline src/app/inner-pages/dev-pages/dev-profile/dev-work-experience/dev-project-card/dev-project-card.component.ts:63]: lifecycle > public methods> private methods

File: src/app/inner-pages/dev-pages/dev-profile/shared/edit-form/edit-form.component.html [hunk 7]
```diff
@@ -0,0 +1,95 @@
+<div [formGroup]="form">
+  <ng-container *ngIf="devProfileService.availableCategories && sectionName === 'categoriesAndSkills'">
```
Comment (timazubar) [inline src/app/inner-pages/dev-pages/dev-profile/shared/edit-form/edit-form.component.html:2]: add sectionNames to enum

File: src/app/inner-pages/dev-pages/dev-profile/shared/edit-form/edit-form.component.ts [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    public devProfileService: DevProfileService,
+    private store: Store<fromCore.State>,
+    private notificationsService: NotificationsService,
+    private userService: UserService
+  ) { }
+
+  ngOnInit(): void {
+    this.initForm();
+    this.store.select(fromCore.getUserInfo)
+    .pipe(first())
```
Comment (timazubar) [inline src/app/inner-pages/dev-pages/dev-profile/shared/edit-form/edit-form.component.ts:42]: отступы

File: src/app/inner-pages/shared/components/developer-resume-full/developer-resume-full.component.ts [hunk 9]
```diff
@@ -46,6 +48,10 @@ export class DeveloperResumeFullComponent implements OnInit {
     });
   }
 
+  public onWorkExperienceClick(id: string) {
```
Comment (timazubar) [inline src/app/inner-pages/shared/components/developer-resume-full/developer-resume-full.component.ts:51]: return type

**Pull request merged**